### PR TITLE
Fix/issue 248

### LINKED ISSUE #248: Able to checkout without triggering required shipment cost
This follows closely this issue https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/issues/179

With PayPal Express, it's possible to set shipping to be calculated on the cart page, and then bypass that setting entirely, thereby never charging shipping even when required.

Shipping is set to add a $12 flat rate to addresses in California and North Carolina.
screen shot 2017-04-03 at 2 37 53 pm
screen shot 2017-04-03 at 2 37 37 pm

Cart with one item. 
screen shot 2017-04-03 at 2 33 55 pm

If I tap on "calculate shipping" it is correctly calculated:
screen shot 2017-04-03 at 2 34 57 pm

If I never click on "calculate shipping" and just go through the yellow PayPal button, I (eventually) arrive at this screen after logging into PayPal on the pop-up:
screen shot 2017-04-03 at 2 40 22 pm

The shipping details are a North Carolina address, and are strictly pulled in from PayPal - I didn't enter them on the WooCommerce store at all.

### PR DESCRIPTION
**Pull request by bor0**
Fixes #248.

Depends on #276.

### CODE REVIEW

**Comment by gedex** [comment]
Please rebase and resolve the conflicts.

**Comment by gedex** [comment]
This PR still includes changes for #276. I'll review once the PR for #276 is merged.

**Comment by bor0** [comment]
Rebased.

File: includes/class-wc-gateway-ppec-cart-handler.php [hunk 1]
```diff
@@ -195,6 +218,8 @@ public function enqueue_scripts() {
 					'locale'      => $settings->get_paypal_locale(),
 					'start_flow'  => esc_url( add_query_arg( array( 'startcheckout' => 'true' ), wc_get_page_permalink( 'cart' ) ) ),
 					'show_modal'  => apply_filters( 'woocommerce_paypal_express_checkout_show_cart_modal', true ),
+					'update_shipping_costs_nonce' => wp_create_nonce( '_wc_ppec_update_shipping_costs_nonce' ),
+					'ajaxurl'     => WC_AJAX::get_endpoint( 'wc_ppec_update_shipping_costs' ),
```
Comment (gedex) [inline includes/class-wc-gateway-ppec-cart-handler.php]: please align the `=>` with the rest of array elements.

**Pull request merged**